### DISCUSSION TITLE
added .ppignore => how to remove these files/folders from DB?

### DISCUSSION THREAD

**Discussion by NOP4** [photoprism/photoprism#1801] [Q&A]
Hello,
I've added some folders and filetypes to .ppignore. Is there any command to indicate photoprism to rescan my photos and remove these folder and files from the database? The "photoprism purge" command don't work here.
If it's not possible, i can only see this solution:
1. Make snapshot of my photo folders (and a physical backup)
2. Manually delete all files and folders corresponding (hundred thousand of files and folders...)
3. Run the "photoprism purge" command
4. Restore the snapshot to recover deleted files
Any idea?

**Comment by graciousgrey** [comment]
"If you are a new user and files or folders have already been indexed, it is generally easiest to reset the database and start with a new index by running photoprism reset in a terminal."
--> see https://docs.photoprism.org/user-guide/library/originals/#ignoring-files-and-folders

**Reply by NOP4** [reply]
No, I'm not a new user. I already have thousands of pictures manually tagged (face recognition errors, ...).
I migrated to a new system, and forgot to reapply my .ppignore to exclude all my RAW files and folders, and the xmp files... Restoring from previous snapshot will be hard in my case.
Is there any script that can clean the database for specific extensions and folders?

**Reply by lastzero** [reply]
We can't offer you a short-term solution that will evaluate all `.ppignore` files in the correct order and generate SQL statements from them, so that your database is magically restored to exactly the state it would have had. Also typos can lead to all sorts of mistakes and drama, for example when your `.ppignore` contains `*`. Should be mentioned in our docs or FAQs as well.

Implementing this concept might be possible and would certainly be a good exercise if we run out of other tasks or get a big check so we can grow our team.

However:

1. You can restore the previous state from a database backup (how did you migrate without one?). PhotoPrism has backup and restore commands if you don't want to use database tools directly.
2. Manually run SQL database queries to remove files and photos you don't want.

**Reply by NOP4** [reply]
I migrated some time ago. Thanks, I'll write SQL queries to clean up database.
I've check the table content and structure, and I need to clean 3 tables: "folders", "files" and "albums". No need cleaning "photos" in my case. I've seen you designed a "deleted_at" field. I'll use it.
I can post my SQL queries here if this can help others.

**Reply by NOP4** [reply]
Hum, looks like I have a bug here in the "folder" table when used with "deleted_at".
Before: folder name "RAW" visible in the Library menu (/library/files/xxxxx) and I can open it. It has an empty content.
After updating entry with "deleted_at" = NOW(), folder is still visible in Library menu, but shows no name, and opening it goes back to root folder (/library/files/).
Is it a bug or am I misusing the deleted_at attribute? Scanning folder for updates does not solve problem. It can be solved by removing deleted_at value for this row.

**Comment by NOP4** [comment]
Do you have an idea regarding the "deleted_at" bug I encounter when using this field? Do I have to open a bug or am I misusing the field?

**Reply by lastzero** [reply]
You just updated deleted_at in the folders table, but not in albums. It's a folder album. Folders alone is only used for directory dropdowns etc.

**Reply by NOP4** [reply]
Hi @lastzero 
I've checked again looking at database content, and this folder is marked as "deleted_at" in the two tables.
Here are the two SQL queries I executed to remove all folders ending with /RAW:
```
UPDATE albums SET deleted_at = NOW()
WHERE album_path like '%/RAW';
```
```
UPDATE folders SET deleted_at = NOW()
WHERE path like '%/RAW';
```
